### ISSUE TITLE
XML bolt-on / mixin support via xmltodict

### ISSUE THREAD

**Issue by skewty** [samuelcolvin/pydantic#952]
# Question 1 

In `xml()` below we `json.loads(self.json())` so we get the serialized values.
Is there a nice way to get the serialized values and avoid going in and out of JSON?

# Question 2

It would be nice to have an `alias` in `Config` (`class BaseConfig` in `main.py`) so it could be referred to instead of `cls.__root_element__` which is minor.  
Is there a nice way to extend `class BaseConfig`?

## Working v.0.32.2 XML support

Posted here in case anyone else is interested.

WARNING: If this is used with v1.00, one should use `__config__.json_loads`.

```py
from typing import Dict, Set, Any, Type, Union, TypeVar, TYPE_CHECKING
from pydantic import BaseModel, StrBytes, Protocol, ValidationError
from pydantic.error_wrappers import ErrorWrapper

import json  # required for v.0.32.2 but not for v.1.00
import xml
import xmltodict  # WARNING!!!  This is not in the standard library

if TYPE_CHECKING:
    Model = TypeVar('Model', bound='BaseModel')
    IntStr = Union[int, str]
    SetIntStr = Set[IntStr]
    DictIntStrAny = Dict[IntStr, Any]


class XMLBaseModel(BaseModel):
    __root_element__ = ''

    @classmethod
    def parse_raw(
        cls: Type['Model'],
        b: StrBytes,
        *,
        content_type: str = None,
        encoding: str = 'utf8',
        proto: Protocol = None,
        allow_pickle: bool = False,
    ) -> 'Model':
        try:
            if content_type and content_type.endswith('xml'):
                b = xmltodict.parse(b, encoding='utf-8' if encoding == 'utf8' else encoding)
                obj = b[cls.__root_element__ or cls.__name__]
                return cls.parse_obj(obj)
        except (ValueError, TypeError, UnicodeDecodeError, xml.parsers.expat.ExpatError) as e:
            raise ValidationError([ErrorWrapper(e, loc='__obj__')], cls)
        return super().parse_raw(
            b, content_type=content_type, encoding=encoding, proto=proto, allow_pickle=allow_pickle
        )

    def xml(
        self,
        *,
        include: Union['SetIntStr', 'DictIntStrAny'] = None,
        exclude: Union['SetIntStr', 'DictIntStrAny'] = None,
        by_alias: bool = False,
        skip_defaults: bool = False,
    ) -> str:
        d = {
            self.__class__.__root_element__ or self.__class__.__name__: json.loads(
                self.json(include=include, exclude=exclude, by_alias=by_alias, skip_defaults=skip_defaults)
            )
        }
        return xmltodict.unparse(d)
```

**Comment by samuelcolvin** [comment]
### Question 1

see #317 

tl;dr, `ujson.loads(self.json())` will likely always be faster 

### Question 2 

I'm not exactly sure what you're asking about I'm afraid. Could you clarify.

### Feature Request

I'm not clear what the feature request is here. Could you describe?
